### DISCUSSION TITLE
Code analysis with Intellisense disabled?

### DISCUSSION THREAD

**Discussion by jalanh11640** [microsoft/vscode-cpptools#11508] [Q&A]
I've configured vscode to use clangd for code completion, etc.  A dialog popped up and said I should disable intellisense, so I did.  I'm now trying to get Code Analysis working.  I've added the following configuration to my *.code-workspace file, but cmd-shift-p doesn't show anything when I type 'Run Code Analysis'.  Does intellisense need to be configured for Code Analysis to work?

	"settings": {
		"clangd.path": "/usr/bin/clangd-17",
		"C_Cpp.codeAnalysis.clangTidy.path": "/usr/bin/clang-tidy-17",
		"C_Cpp.codeAnalysis.clangTidy.enabled": true,
		"C_Cpp.codeAnalysis.clangTidy.checks.enabled": ["modernize-*"],
		"C_Cpp.codeAnalysis.clangTidy.useBuildPath": true,
		"C_Cpp.default.compileCommands": "${workspaceFolder}/build/compile_commands.json"
	}

Thanks,
Jeff

**Comment by bobbrow** [comment]
> Does intellisense need to be configured for Code Analysis to work?

Yes, it does.  We reuse the IntelliSense configurations for code analysis, so if you've disabled IntelliSense the only thing you get from our extension is the debugger.

**Reply by jalanh11640** [reply]
Thanks for responding. That's unfortunate that you can't use the Code Analysis if Intellisense is disabled.  Any chance this could be changed?  I personally would like to use the Code Analysis and clangd.

My issue with intellisense is it is INCREDIBLY slow.  When its done, it seems to work great, but then seems to get corrupted or something and I have to 'Reset Intellisense' to get it back on track (after another SLOW crawl through all the headers)

**Reply by sean-mcmanus** [reply]
@jalanh11640 We're tracking that with https://github.com/microsoft/vscode-cpptools/issues/8874 . The workaround is to add a tasks.json entry that calls clang-tidy directly (or run it via the command line, etc.).
